packages/frontend/core/src/components/affine/setting-modal/general-setting/plans/ai/ai-plan.tsx

```diff
@@ -0,0 +1,102 @@
+import { useCurrentLoginStatus } from '@affine/core/hooks/affine/use-current-login-status';
+import {
+  type SubscriptionMutator,
+  useUserSubscription,
+} from '@affine/core/hooks/use-subscription';
+import { timestampToLocalDate } from '@affine/core/utils';
+import {
+  type PricesQuery,
+  SubscriptionPlan,
+  SubscriptionRecurring,
+} from '@affine/graphql';
+
+import { AIPlanLayout } from '../layout';
+import * as styles from './ai-plan-card.css';
+import { AICancel } from './cancel';
+import { AILogin } from './login';
+import { AIResume } from './resume';
+import { AISubscribe } from './subscribe';
+import type { BaseActionProps } from './types';
+
+interface AIPlanProps {
+  price?: PricesQuery['prices'][number];
+  onSubscriptionUpdate: SubscriptionMutator;
+}
+export const AIPlan = ({ price, onSubscriptionUpdate }: AIPlanProps) => {
+  const plan = SubscriptionPlan.AI;
+  const recurring = SubscriptionRecurring.Yearly;
+
+  const loggedIn = useCurrentLoginStatus() === 'authenticated';
+
+  const [subscription] = useUserSubscription(plan);
+
+  // yearly subscription should always be available
+  if (!price?.yearlyAmount) return null;
+
+  const baseActionProps: BaseActionProps = {
+    plan,
+    price,
+    recurring,
+    onSubscriptionUpdate,
+  };
+  // const isCancelled = subscription.status === SubscriptionStatus.Canceled;
+  const isCancelled = !!subscription?.canceledAt;
```

Comment:
`canceledAt` is the date user cancel the subscription recurring instead of the end of current period

Reply:
It is used here to judge whether the current subscription has been cancelled. I'm not quite sure whether using `subscription.status === SubscriptionStatus.Canceled` will be more accurate?

Reply:
`Status.Canceled` means user is no longer available using that subscription.
`sub.canceledAt` means user stopped further subscription renew.

so `sub.canceledAt` here is the right one, you can remove the comment